### ISSUE TITLE
TextParser errors after update 4.0 -> 4.1 -> 4.2

### ISSUE THREAD

**Issue by vovpff** [YetiForceCompany/YetiForceCRM#5278]
Hello one more time. History of this error here #5171
Yesterday I established new server based on Ubuntu 16.04.
Migration from WAMP to LAMP was successful without any errors.
This is v4.0 was established. After that I made step by step update 4.0 to 4.1 (upd pack v6) and v 4.1 to 4.2 (upd pack v1). At first sight all works well but when I try to change subject of IGDN from summary or full edit page I has endless loading or blank page (maybe this is 500 error).
I checked log from /cache/logs/phperrors.log and found this stuff.

```
[26-Sep-2017 23:22:36 Europe/Moscow] PHP Fatal error:  Uncaught TypeError: Argument 1 passed to App\TextParser::getDisplayValueByField() must be an instance of Vtiger_Field_Model, string given, called in /var/www/html/my40/vendor/yetiforce/TextParser.php on line 413 and defined in /var/www/html/my40/vendor/yetiforce/TextParser.php:633
Stack trace:
#0 /var/www/html/my40/vendor/yetiforce/TextParser.php(413): App\TextParser->getDisplayValueByField('Gaiser T8', Object(IGDN_Field_Model))
#1 /var/www/html/my40/vendor/yetiforce/TextParser.php(243): App\TextParser->record('ChangesListChan...')
#2 [internal function]: App\TextParser->App\{closure}(Array)
#3 /var/www/html/my40/vendor/yetiforce/TextParser.php(246): preg_replace_callback('/\\$\\((\\w+) : ([...', Object(Closure), '<meta charset="...')
#4 /var/www/html/my40/modules/com_vtiger_workflow/tasks/VTEmailTask.php(60): App\TextParser->parse()
#5 /var/www/html/my40/modules/com_vtiger_workflow/VTWorkflowManager.php(526): VTEmailTask->doTask(Object(IGDN_Record_Model))
#6 /var/www/html/my40/modules/Vtiger/handlers/Workflo in /var/www/html/my40/vendor/yetiforce/TextParser.php on line 633
[26-Sep-2017 23:22:59 Europe/Moscow] PHP Fatal error:  Uncaught TypeError: Argument 1 passed to App\TextParser::getDisplayValueByField() must be an instance of Vtiger_Field_Model, string given, called in /var/www/html/my40/vendor/yetiforce/TextParser.php on line 413 and defined in /var/www/html/my40/vendor/yetiforce/TextParser.php:633
Stack trace:
#0 /var/www/html/my40/vendor/yetiforce/TextParser.php(413): App\TextParser->getDisplayValueByField('Gaiser T8 8888', Object(IGDN_Field_Model))
#1 /var/www/html/my40/vendor/yetiforce/TextParser.php(243): App\TextParser->record('ChangesListChan...')
#2 [internal function]: App\TextParser->App\{closure}(Array)
#3 /var/www/html/my40/vendor/yetiforce/TextParser.php(246): preg_replace_callback('/\\$\\((\\w+) : ([...', Object(Closure), '<meta charset="...')
#4 /var/www/html/my40/modules/com_vtiger_workflow/tasks/VTEmailTask.php(60): App\TextParser->parse()
#5 /var/www/html/my40/modules/com_vtiger_workflow/VTWorkflowManager.php(526): VTEmailTask->doTask(Object(IGDN_Record_Model))
#6 /var/www/html/my40/modules/Vtiger/handlers/Wo in /var/www/html/my40/vendor/yetiforce/TextParser.php on line 633
[26-Sep-2017 23:29:31 Europe/Moscow] PHP Fatal error:  Uncaught TypeError: Argument 1 passed to App\TextParser::getDisplayValueByField() must be an instance of Vtiger_Field_Model, string given, called in /var/www/html/my40/vendor/yetiforce/TextParser.php on line 413 and defined in /var/www/html/my40/vendor/yetiforce/TextParser.php:633
Stack trace:
#0 /var/www/html/my40/vendor/yetiforce/TextParser.php(413): App\TextParser->getDisplayValueByField('TPT HB05 \xD0\x9C\xD0\x9D\xD0\x98...', Object(IGDN_Field_Model))
#1 /var/www/html/my40/vendor/yetiforce/TextParser.php(243): App\TextParser->record('ChangesListChan...')
#2 [internal function]: App\TextParser->App\{closure}(Array)
#3 /var/www/html/my40/vendor/yetiforce/TextParser.php(246): preg_replace_callback('/\\$\\((\\w+) : ([...', Object(Closure), '<meta charset="...')
#4 /var/www/html/my40/modules/com_vtiger_workflow/tasks/VTEmailTask.php(60): App\TextParser->parse()
#5 /var/www/html/my40/modules/com_vtiger_workflow/VTWorkflowManager.php(526): VTEmailTask->doTask(Object(IGDN_Record_Model))
#6 /var/www/html/my40/modu in /var/www/html/my40/vendor/yetiforce/TextParser.php on line 633
```

There are imaged 3 attempts by next order: editing field in summary view page, editing field at full record edit page, editing field in summary view page

P.S. I m not sure for 100% but I noticed what error appears for Logistics modules. I can check it one more time this evening.

**Comment by vovpff** [comment]
@mariuszkrzaczkowski I'm so sorry for confusion.
I checked a lot of modules and found what only IGDN module calls 500 error.
I checked my CRM and notice what only this module has active Workflow to send email if record changed.
Looks like problem was in message source. My source is:

```
<meta charset="UTF-8">Расходный ордер <a href="$(record : CrmDetailViewURL)$">$(record : RecordLabel)$</a>&nbsp;для клиента $(record : accountid)$&nbsp;создан либо изменен&nbsp;<br />
<br />
Последние изменения:<br />
$(record : ChangesListChanges)$
```

After removing `<meta charset="UTF-8">` and link rebuilding system works well!

One more time sorry for your time.
